blockchain/chain.go

```diff
@@ -1009,6 +999,28 @@ func (b *BlockChain) disconnectBlock(node *blockNode, block, parent *dcrutil.Blo
 	return nil
 }
 
+// watchAndUpdateChainIndex observes chain index state and indexes, de-indexes blocks
+func (b *BlockChain) watchAndUpdateChainIndex() {
+	// run on initial startup
+	err := b.indexManager.CatchUpIndexes(b, b.interrupt)
+	if err != nil && err.Error() != errInterruptRequested.Error() {
```

Comment:
No need to `err.Error()`.  Just check `err != errInterruptRequested`.

Reply:
it didn't work somehow with `err != errInterruptRequested`

Reply:
The error in the indexers package is not the same as the one in the blockchain package, although they are named the same.  Since you're trying to detect the error out of the indexers package, you'd need to export it and check against that.

e.g. `if err != indexers.ErrInterruptRequested`.

That said, as mentioned elsewhere, the idea of panic on non-database errors, or internal assertions, from outside of the indexers package is not something that dcrd can do.  A failure to index a block should not take down the entire process.